docs/integrations/databases/supabase.mdx

```diff
@@ -3,170 +3,666 @@ title: Integrate Supabase with Clerk
+      setPosts(error || data);
+    };
+
+    const inputRef = useRef<HTMLInputElement>(null);
+    // Sends the value of a text input as a post
+    const sendPost = async () => {
+      if (!inputRef.current?.value) return;
+      const client = await getSupabaseClient();
+      const { data, error } = await client.from("Posts").insert({
+        user_id: userId,
+        // Replace content with whatever field you want
+        content: inputRef.current?.value,
+      });
+    };
+    ```
+    </CodeBlockTabs>
+
+1. Finally, edit your component to return a basic UI that allows you to list all your posts and send new ones:
+    <CodeBlockTabs type="framework" options={["Next.js App Router", "Next.js Pages Router", "React"]}>
+    ```tsx filename="app/supabase/page.tsx"
+    return (
+      <>
+        <div style={{ display: "flex", flexDirection: "column" }}>
+          <input style={{ color: "black" }} type="text" ref={inputRef} />
+          <button onClick={sendPost}>Send Post</button>
+          <button onClick={listPosts}>Fetch Posts</button>
+        </div>
+        <h2>Posts</h2>
+        <p>
+          {!posts ? (
+            <strong>No posts</strong>
+          ) : (
+            <ul>
+              {posts.map((post: any) => (
+                <li key={post.id}>{post.content}</li>
+              ))}
+            </ul>
+          )}
+        </p>
+      </>
+    );
+    ```
+    ```tsx filename="page/supabase/index.tsx"
+    return (
+      <>
+        <div style={{ display: "flex", flexDirection: "column" }}>
+          <input style={{ color: "black" }} type="text" ref={inputRef} />
+          <button onClick={sendPost}>Send Post</button>
+          <button onClick={listPosts}>Fetch Posts</button>
+        </div>
+        <h2>Posts</h2>
+        <p>
+          {!posts ? (
+            <strong>No posts</strong>
+          ) : (
+            <ul>
+              {posts.map((post: any) => (
+                <li key={post.id}>{post.content}</li>
+              ))}
+            </ul>
+          )}
+        </p>
+      </>
+    );
+    ```
+    ```tsx filename="components/supabase.tsx"
+    return (
+      <>
+        <div style={{ display: "flex", flexDirection: "column" }}>
+          <input style={{ color: "black" }} type="text" ref={inputRef} />
+          <button onClick={sendPost}>Send Post</button>
+          <button onClick={listPosts}>Fetch Posts</button>
+        </div>
+        <h2>Posts</h2>
+        <p>
+          {!posts ? (
+            <strong>No posts</strong>
+          ) : (
+            <ul>
+              {posts.map((post: any) => (
+                <li key={post.id}>{post.content}</li>
+              ))}
+            </ul>
+          )}
+        </p>
+      </>
+    );
+    ```
+    </CodeBlockTabs>
+1. The final result should be similar to this:
+    <CodeBlockTabs type="framework" options={["Next.js App Router", "Next.js Pages Router", "React"]}>
+    ```ts filename="app/supabase/page.tsx"
+    "use client";
```

Comment:
Glad we're making the distinction from above here -- may not need the callout above, but still something to think about